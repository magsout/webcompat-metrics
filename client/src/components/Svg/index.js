/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import SVGInline from "react-svg-inline";
import PropTypes from "prop-types";

const Svg = props => {
  return (
    <SVGInline
      cleanup={[
        "title",
        "desc",
        "comment",
        "defs",
        "sketchMSShape",
        "sketchMSPage",
        "sketchMSLayerGroup",
      ]}
      {...props}
    />
  );
};

Svg.propTypes = {
  svg: PropTypes.string.isRequired,
  width: PropTypes.string,
  height: PropTypes.string,
};

export default Svg;
