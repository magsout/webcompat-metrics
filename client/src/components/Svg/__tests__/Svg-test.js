/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import Svg from "../";
import SVGTest from "../../../assets/svg/Bugdiagnosis.svg";

it("renders Svg default correctly", () => {
  const tree = renderer
    .create(<Svg width="2em" height="2em" svg={SVGTest} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
