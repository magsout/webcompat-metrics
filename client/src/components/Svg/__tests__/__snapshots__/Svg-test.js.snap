// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Svg default correctly 1`] = `
<span
  className="SVGInline SVGInline--cleaned"
  dangerouslySetInnerHTML={
    Object {
      "__html": "test-file-stub",
    }
  }
/>
`;
